### PR TITLE
Implemented change password functionality

### PR DESCRIPTION
**Pull request by ramadis**
## Summary

Implemented change password functionality

## Test Plan

* Log into your account
* Go to profile
* Click on Change password
* Update password
* Log in again with your new password

## Trello Card 

 https://trello.com/c/F8bGzo6i

## Screenshot

matchpassword

## Notes

Change password forces user to re-login

### CODE REVIEW

File: src/app/app.routes.js [hunk 1]
```diff
@@ -100,6 +100,16 @@ angular.module('app').config([
             templateUrl: '../app/components/private/suggestion.html'
           }
         }
+      })
+      .state('private.changePwd', {
+        url: '/newpwd',
```
Comment (JISevilla92) [inline src/app/app.routes.js]: Remember that the url is what users end seeing and `newpwd` might not be clear enough.

File: src/app/components/private/password.jade [hunk 2]
```diff
@@ -0,0 +1,12 @@
+h2
+  {{ 'UPDATEPWD.TITLE' | translate }}
+
+form.form-register(name='changepwdForm' ng-submit='profileCtrl.updatePwd(profileCtrl.password)')
+  fieldset
+    input(type='password' ng-model='profileCtrl.password.new' ng-minlength='8' placeholder="{{ 'UPDATEPWD.NEW' | translate }}" required)
+  fieldset
+    input(type='password' ng-model='profileCtrl.password.repeat' ng-minlength='8' placeholder="{{ 'UPDATEPWD.REPEAT' | translate }}" required)
+  fieldset
+    input.btn.btn-primary(type='submit' ng-disabled='changepwdForm.$invalid || profileCtrl.password.repeat != profileCtrl.password.new' value="{{ 'UPDATEPWD.BUTTON' | translate }}")
```
Comment (JISevilla92) [inline src/app/components/private/password.jade]: Small comment: You could use a directive here (https://docs.angularjs.org/guide/directive). They are really powerful and a big part of what Angular brings to the table. Takes a while to embrace them (At least for me) so don't worry.

File: src/app/components/private/profile.jade [hunk 3]
```diff
@@ -26,3 +26,5 @@ form.form-register(name="profileEdit" ng-show='profileCtrl.editable' ng-submit='
 
 button.btn.btn-primary(ng-click='profileCtrl.editable = true' ng-hide='profileCtrl.editable')
   {{ 'PROFILE.EDITBUTTON' | translate }}
+button.btn.btn-danger(ng-hide='profile.editable' ui-sref='private.changePwd')
```
Comment (JISevilla92) [inline src/app/components/private/profile.jade]: `profileCtrl`

File: src/app/services/bookService.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ angular.module('app').factory('bookService', [
   function (Restangular) {
     return {
       addSuggestion: (book, success, failure) => {
-          Restangular.one('classes', 'BookSuggestions').post(book).then(success, failure);
+        Restangular.one('classes', 'BookSuggestions').post(book).then(success, failure);
```
Comment (JISevilla92) [inline src/app/services/bookService.js]: Use `.all()` to make reference to a collection.

File: src/app/components/private/profile.jade (hunk 3 above)
Comment (sbalay) [inline src/app/components/private/profile.jade]: :arrow_up:

File: src/app/components/private/profileController.js [hunk 5]
```diff
@@ -14,5 +15,12 @@ angular.module('app').controller('ProfileController', [
         this.editable = false;
       });
     };
+
+    this.updatePwd = function (pwd) {
+      userService.updatePassword(pwd.repeat);
+      authenticationService.clearCredentials();
```
Comment (sbalay) [inline src/app/components/private/profileController.js]: Do this and the $state.go in the success callback of update password

**Comment by sbalay** [comment]
:+1:

**Pull request merged**